test/unit/language-extensions/__snapshots__/multi.spec.ts.snap

```diff
@@ -54,24 +58,78 @@ exports[`invalid $multi call (({ $multi });): code 1`] = `"local ____ = nil"`;
 
 exports[`invalid $multi call (({ $multi });): diagnostics 1`] = `"main.ts(2,12): error TSTL: The $multi function must be called in an expression that is returned."`;
 
-exports[`invalid $multi call (const [a = 0] = $multi()): code 1`] = `""`;
+exports[`invalid $multi call (const [a = 0] = $multi()): code 1`] = `
+"a = ____(_G)
+if a == nil then
+    a = 0
+end"
+`;
+
+exports[`invalid $multi call (const [a = 0] = $multi()): diagnostics 1`] = `"main.ts(2,25): error TSTL: The $multi function must be called in an expression that is returned."`;
+
+exports[`invalid $multi call (const {} = $multi();): code 1`] = `
+"local ____ = {
+    ____(_G)
+}"
+`;
+
+exports[`invalid $multi call (const {} = $multi();): diagnostics 1`] = `"main.ts(2,20): error TSTL: The $multi function must be called in an expression that is returned."`;
+
+exports[`invalid $multi call (const a = $multi();): code 1`] = `
+"a = {
+    ____(_G)
+}"
+`;
+
+exports[`invalid $multi call (const a = $multi();): diagnostics 1`] = `"main.ts(2,19): error TSTL: The $multi function must be called in an expression that is returned."`;
+
+exports[`invalid $multi implicit cast: code 1`] = `
+"function badMulti(self)
+    return \\"foo\\", 42
+end"
+`;
 
-exports[`invalid $multi call (const [a = 0] = $multi()): diagnostics 1`] = `"main.ts(2,15): error TSTL: This array binding pattern cannot have initializers."`;
+exports[`invalid $multi implicit cast: diagnostics 1`] = `"main.ts(3,20): error TSTL: The $multi function can only be used to return a LuaMultiReturn type."`;
```

Comment:
Would `The $multi function cannot be cast to a non-LuaMultiReturn type.` be clearer or am I forgetting some cases for this diagnostic here?

Reply:
Sounds good to me.